playground/src/utils/helper_functions.tsx

```diff
@@ -139,22 +141,41 @@ Did you prove the program first?`;
 }
 
 /**
- * Helper function to format the debug output.
+ * Helper function to get next power of 2.
  */
-export function addNewlineAfterWhitespace(str: string): string {
-  let inSquareBrackets = false;
-  let result = '';
-  for (let i = 0; i < str.length; i++) {
-    if (str[i] === '['  || str[i] === '{') {
-      inSquareBrackets = true;
-    } else if (str[i] === ']' || str[i] === '}') {
-      inSquareBrackets = false;
-    }
-    if (!inSquareBrackets && /\s/.test(str[i])) {
-      result += '\n';
-    } else {
-      result += str[i];
+
+export function nextPowerOf2(x: number): number {
+  let power = 1;
+    while (power < x) {
+      power <<= 1;
     }
-  }
-  return result;
+  return Math.max(power, 1024);
+} 
+
+/**
+ * Helper function to format the Debugger output.
+ */
+export function formatDebuggerOutput(debugOutput: DebugOutput): string {
+  const output = 
```

Comment:
nit: this looks like JSON, but it is not. Two possible improvements:

- Just print the JSON
- remove the quotes and show elements as a tidy table, i.e.:

```
clock: 5
stack: [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
instruction: push.1
operation: Some(Incr)
operation index: 2
operation count: 2
memory: 
```